src/System.Text.Json/src/System/Text/Json/Node/JsonObjectEnumerator.cs

```diff
@@ -8,38 +8,63 @@ namespace System.Text.Json
     /// </summary>
     public struct JsonObjectEnumerator : IEnumerator<KeyValuePair<string, JsonNode>>
     {
-        private Dictionary<string, JsonNode>.Enumerator _enumerator;
+        private JsonObjectProperty _first;
+        private JsonObjectProperty _current;
 
         /// <summary>
         ///   Initializes a new instance of the <see cref="JsonObjectEnumerator"/> class supporting an interation over provided JSON object.
         /// </summary>
         /// <param name="jsonObject">JSON object to iterate over.</param>
-        public JsonObjectEnumerator(JsonObject jsonObject) => _enumerator = jsonObject._dictionary.GetEnumerator();
+        public JsonObjectEnumerator(JsonObject jsonObject)
+        {
+            _first = jsonObject._first;
+            _current = null;
+        }
 
         /// <summary>
         ///   Gets the property in the JSON object at the current position of the enumerator.
         /// </summary>
-        public KeyValuePair<string, JsonNode> Current => _enumerator.Current;
+        public KeyValuePair<string, JsonNode> Current => new KeyValuePair<string, JsonNode>(_current._name, _current._value);
```

Comment:
should this `return default` when _current is null? (before first MoveNext, after last one?)

Reply:
Ok, I will fix it.